diff --git a/build/make/gen_msvs_vcxproj.sh b/build/make/gen_msvs_vcxproj.sh
index 916851662..e60405bc9 100755
--- a/build/make/gen_msvs_vcxproj.sh
+++ b/build/make/gen_msvs_vcxproj.sh
@@ -394,7 +394,7 @@ generate_vcxproj() {
               else
                 config_suffix=""
               fi
-              tag_content TargetName "${name}${lib_sfx}${config_suffix}"
+              tag_content TargetName "${name}"
             fi
             close_tag PropertyGroup
         done
