diff --git a/flutter/android/app/build.gradle b/flutter/android/app/build.gradle
index f4dc69e..6b835fd 100644
--- a/flutter/android/app/build.gradle
+++ b/flutter/android/app/build.gradle
@@ -67,6 +67,19 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules'
         }
     }
+
+    applicationVariants.all { variant ->
+        variant.outputs.each { output ->
+            output.processManifest.doLast { task ->
+                def outputDir = multiApkManifestOutputDirectory.asFile.get()
+                File manifestOutFile = new File(outputDir, "AndroidManifest.xml")
+                if (manifestOutFile.exists()) {
+                    def newFileContents = manifestOutFile.getText('UTF-8').replace("android:debuggable=\"true\"", "")
+                    manifestOutFile.write(newFileContents, 'UTF-8')
+                }
+            }
+        }
+    }
 }
 
 flutter {
