diff --git a/libavcodec/amfenc.h b/libavcodec/amfenc.h
--- a/libavcodec/amfenc.h
+++ b/libavcodec/amfenc.h
@@ -51,6 +51,7 @@ typedef struct AmfContext {
     int                 hwsurfaces_in_queue;
     int                 hwsurfaces_in_queue_max;
     int                 query_timeout_supported;
+    int64_t             query_timeout;

     // helpers to handle async calls
     int                 delayed_drain;
diff --git a/libavcodec/amfenc_h264.c b/libavcodec/amfenc_h264.c
--- a/libavcodec/amfenc_h264.c
+++ b/libavcodec/amfenc_h264.c
@@ -141,6 +141,7 @@ static const AVOption options[] = {

     { "smart_access_video",     "Enable Smart Access Video to enhance  performance by utilizing both APU and dGPU memory access",    OFFSET(smart_access_video), AV_OPT_TYPE_BOOL, {.i64 = -1  }, -1, 1, VE},

+    { "query_timeout", "Timeout for QueryOutput call in ms", OFFSET(query_timeout), AV_OPT_TYPE_INT64, { .i64 = -1 }, -1, 1000, VE },
     //Pre Analysis options
     { "preanalysis",                            "Enable preanalysis",                                           OFFSET(preanalysis),                            AV_OPT_TYPE_BOOL,   {.i64 = -1 }, -1, 1, VE },

@@ -541,7 +542,8 @@ static av_cold int amf_encode_init_h264(AVCodecContext *avctx)
     }

     // Wait inside QueryOutput() if supported by the driver
-    AMF_ASSIGN_PROPERTY_INT64(res, ctx->encoder, AMF_VIDEO_ENCODER_QUERY_TIMEOUT, 1);
+    AMF_ASSIGN_PROPERTY_INT64(res, ctx->encoder, AMF_VIDEO_ENCODER_QUERY_TIMEOUT,
+                              ctx->query_timeout >= 0 ? ctx->query_timeout : 1);
     res = ctx->encoder->pVtbl->GetProperty(ctx->encoder, AMF_VIDEO_ENCODER_QUERY_TIMEOUT, &var);
     ctx->query_timeout_supported = res == AMF_OK && var.int64Value;

diff --git a/libavcodec/amfenc_hevc.c b/libavcodec/amfenc_hevc.c
--- a/libavcodec/amfenc_hevc.c
+++ b/libavcodec/amfenc_hevc.c
@@ -111,6 +111,7 @@ static const AVOption options[] = {

     { "smart_access_video",     "Enable Smart Access Video to enhance  performance by utilizing both APU and dGPU memory access",        OFFSET(smart_access_video), AV_OPT_TYPE_BOOL, {.i64 = -1  }, -1, 1, VE},

+    { "query_timeout", "Timeout for QueryOutput call in ms", OFFSET(query_timeout), AV_OPT_TYPE_INT64, { .i64 = -1 }, -1, 1000, VE },
     //Pre Analysis options
     { "preanalysis",                            "Enable preanalysis", OFFSET(preanalysis),                            AV_OPT_TYPE_BOOL,   {.i64 = -1 }, -1, 1, VE },

@@ -459,7 +460,8 @@ static av_cold int amf_encode_init_hevc(AVCodecContext *avctx)
     }

     // Wait inside QueryOutput() if supported by the driver
-    AMF_ASSIGN_PROPERTY_INT64(res, ctx->encoder, AMF_VIDEO_ENCODER_HEVC_QUERY_TIMEOUT, 1);
+    AMF_ASSIGN_PROPERTY_INT64(res, ctx->encoder, AMF_VIDEO_ENCODER_HEVC_QUERY_TIMEOUT,
+                              ctx->query_timeout >= 0 ? ctx->query_timeout : 1);
     res = ctx->encoder->pVtbl->GetProperty(ctx->encoder, AMF_VIDEO_ENCODER_HEVC_QUERY_TIMEOUT, &var);
     ctx->query_timeout_supported = res == AMF_OK && var.int64Value;

