diff --git a/libavcodec/amfenc.c b/libavcodec/amfenc.c
--- a/libavcodec/amfenc.c
+++ b/libavcodec/amfenc.c
@@ -321,6 +321,7 @@ int ff_amf_encode_init(AVCodecContext *avctx)
     ctx->dts_delay = 0;
 
     ctx->hwsurfaces_in_queue = 0;
+    ctx->av_bitrate = avctx->bit_rate;
 
     if (avctx->hw_device_ctx) {
         hwdev_ctx = (AVHWDeviceContext*)avctx->hw_device_ctx->data;
@@ -670,6 +671,23 @@ static AMF_RESULT amf_query_output(AVCodecContext *avctx, AMFBuffer **buffer)
     return ret;
 }
 
+static void reconfig_encoder(AVCodecContext *avctx)
+{
+    AMFEncoderContext *ctx = avctx->priv_data;
+    AMF_RESULT res = AMF_OK;
+
+    if (ctx->av_bitrate == avctx->bit_rate)
+        return;
+
+    av_log(ctx, AV_LOG_INFO, "change bitrate from %" PRId64 " to %" PRId64 "\n",
+           ctx->av_bitrate, avctx->bit_rate);
+    ctx->av_bitrate = avctx->bit_rate;
+    if (avctx->codec->id == AV_CODEC_ID_H264)
+        AMF_ASSIGN_PROPERTY_INT64(res, ctx->encoder, AMF_VIDEO_ENCODER_TARGET_BITRATE, avctx->bit_rate);
+    else if (avctx->codec->id == AV_CODEC_ID_HEVC)
+        AMF_ASSIGN_PROPERTY_INT64(res, ctx->encoder, AMF_VIDEO_ENCODER_HEVC_TARGET_BITRATE, avctx->bit_rate);
+}
+
 int ff_amf_receive_packet(AVCodecContext *avctx, AVPacket *avpkt)
 {
     AMFEncoderContext     *ctx = avctx->priv_data;
@@ -684,6 +702,8 @@ int ff_amf_receive_packet(AVCodecContext *avctx, AVPacket *avpkt)
     int64_t     pts = 0;
     int max_b_frames = ctx->max_b_frames < 0 ? 0 : ctx->max_b_frames;
 
+    reconfig_encoder(avctx);
+
     if (!ctx->encoder){
         av_frame_free(&frame);
         return AVERROR(EINVAL);
diff --git a/libavcodec/amfenc.h b/libavcodec/amfenc.h
--- a/libavcodec/amfenc.h
+++ b/libavcodec/amfenc.h
@@ -101,6 +101,7 @@ typedef struct AMFEncoderContext {
     int                 qvbr_quality_level;
     int                 hw_high_motion_quality_boost;
     int                 forced_idr;
+    int64_t             av_bitrate;
 
     // HEVC - specific options
 
