diff --git a/libavcodec/mediacodec_wrapper.c b/libavcodec/mediacodec_wrapper.c
index 283bbe72..11625ff7 100644
--- a/libavcodec/mediacodec_wrapper.c
+++ b/libavcodec/mediacodec_wrapper.c
@@ -35,6 +35,8 @@
 #include "ffjni.h"
 #include "mediacodec_wrapper.h"

+#define PARAMETER_KEY_VIDEO_BITRATE "video-bitrate"
+
 struct JNIAMediaCodecListFields {

     jclass mediacodec_list_class;
@@ -194,6 +196,7 @@ struct JNIAMediaCodecFields {

     jmethodID set_input_surface_id;
     jmethodID signal_end_of_input_stream_id;
+    jmethodID set_parameters_id;

     jclass mediainfo_class;

@@ -247,6 +250,7 @@ static const struct FFJniField jni_amediacodec_mapping[] = {

         { "android/media/MediaCodec", "setInputSurface", "(Landroid/view/Surface;)V", FF_JNI_METHOD, OFFSET(set_input_surface_id), 0 },
         { "android/media/MediaCodec", "signalEndOfInputStream", "()V", FF_JNI_METHOD, OFFSET(signal_end_of_input_stream_id), 0 },
+        { "android/media/MediaCodec", "setParameters", "(Landroid/os/Bundle;)V", FF_JNI_METHOD, OFFSET(set_parameters_id), 0 },

     { "android/media/MediaCodec$BufferInfo", NULL, NULL, FF_JNI_CLASS, OFFSET(mediainfo_class), 1 },

@@ -292,6 +296,21 @@ typedef struct FFAMediaCodecJni {

 static const FFAMediaCodec media_codec_jni;

+struct JNIABundleFields {
+    jclass bundle_class;
+    jmethodID init_id;
+    jmethodID put_int_id;
+};
+
+#define OFFSET(x) offsetof(struct JNIABundleFields, x)
+static const struct FFJniField jni_abundle_mapping[] = {
+    { "android/os/Bundle", NULL, NULL, FF_JNI_CLASS, OFFSET(bundle_class), 1 },
+    { "android/os/Bundle", "<init>", "()V", FF_JNI_METHOD, OFFSET(init_id), 1 },
+    { "android/os/Bundle", "putInt", "(Ljava/lang/String;I)V", FF_JNI_METHOD, OFFSET(put_int_id), 1 },
+    { NULL }
+};
+#undef OFFSET
+
 #define JNI_GET_ENV_OR_RETURN(env, log_ctx, ret) do {              \
     (env) = ff_jni_get_env(log_ctx);                               \
     if (!(env)) {                                                  \
@@ -1762,6 +1781,59 @@ static int mediacodec_jni_signalEndOfInputStream(FFAMediaCodec *ctx)
     return 0;
 }

+static int mediacodec_jni_setParameter(FFAMediaCodec *ctx, const char *name, int value)
+{
+    JNIEnv *env = NULL;
+    struct JNIABundleFields jfields = { 0 };
+    jobject object = NULL;
+    jstring key = NULL;
+    FFAMediaCodecJni *codec = (FFAMediaCodecJni *)ctx;
+    void *log_ctx = codec;
+    int ret = AVERROR_EXTERNAL;
+
+    JNI_GET_ENV_OR_RETURN(env, codec, AVERROR_EXTERNAL);
+
+    if (ff_jni_init_jfields(env, &jfields, jni_abundle_mapping, 0, log_ctx) < 0) {
+        av_log(log_ctx, AV_LOG_ERROR, "Failed to init jfields\n");
+        goto fail;
+    }
+    object = (*env)->NewObject(env, jfields.bundle_class, jfields.init_id);
+    if (!object) {
+        av_log(log_ctx, AV_LOG_ERROR, "Failed to create bundle object\n");
+        goto fail;
+    }
+    key = ff_jni_utf_chars_to_jstring(env, name, log_ctx);
+    if (!key) {
+        av_log(log_ctx, AV_LOG_ERROR, "Failed to convert key to jstring\n");
+        goto fail;
+    }
+    (*env)->CallVoidMethod(env, object, jfields.put_int_id, key, value);
+    if (ff_jni_exception_check(env, 1, log_ctx) < 0)
+        goto fail;
+
+    if (!codec->jfields.set_parameters_id) {
+        av_log(log_ctx, AV_LOG_ERROR, "System doesn't support setParameters\n");
+        goto fail;
+    }
+    (*env)->CallVoidMethod(env, codec->object, codec->jfields.set_parameters_id, object);
+    if (ff_jni_exception_check(env, 1, log_ctx) < 0)
+        goto fail;
+    ret = 0;
+
+fail:
+    if (key)
+        (*env)->DeleteLocalRef(env, key);
+    if (object)
+        (*env)->DeleteLocalRef(env, object);
+    ff_jni_reset_jfields(env, &jfields, jni_abundle_mapping, 0, log_ctx);
+    return ret;
+}
+
+static int mediacodec_jni_setDynamicBitrate(FFAMediaCodec *ctx, int bitrate)
+{
+    return mediacodec_jni_setParameter(ctx, PARAMETER_KEY_VIDEO_BITRATE, bitrate);
+}
+
 static int mediacodec_jni_setAsyncNotifyCallback(FFAMediaCodec *codec,
                                                  const FFAMediaCodecOnAsyncNotifyCallback *callback,
                                                  void *userdata)
@@ -1829,6 +1901,7 @@ static const FFAMediaCodec media_codec_jni = {
     .getConfigureFlagEncode = mediacodec_jni_getConfigureFlagEncode,
     .cleanOutputBuffers = mediacodec_jni_cleanOutputBuffers,
     .signalEndOfInputStream = mediacodec_jni_signalEndOfInputStream,
+    .setDynamicBitrate = mediacodec_jni_setDynamicBitrate,
     .setAsyncNotifyCallback = mediacodec_jni_setAsyncNotifyCallback,
 };

@@ -2376,6 +2449,12 @@ static int mediacodec_ndk_signalEndOfInputStream(FFAMediaCodec *ctx)
     return 0;
 }

+static int mediacodec_ndk_setDynamicBitrate(FFAMediaCodec *ctx, int bitrate)
+{
+    av_log(ctx, AV_LOG_ERROR, "ndk setDynamicBitrate unavailable\n");
+    return AVERROR(ENOSYS);
+}
+
 static void mediacodec_ndk_onInputAvailable(AMediaCodec *impl, void *userdata,
                                             int32_t index)
 {
@@ -2525,6 +2604,7 @@ static const FFAMediaCodec media_codec_ndk = {
     .getConfigureFlagEncode = mediacodec_ndk_getConfigureFlagEncode,
     .cleanOutputBuffers = mediacodec_ndk_cleanOutputBuffers,
     .signalEndOfInputStream = mediacodec_ndk_signalEndOfInputStream,
+    .setDynamicBitrate = mediacodec_ndk_setDynamicBitrate,
     .setAsyncNotifyCallback = mediacodec_ndk_setAsyncNotifyCallback,
 };

diff --git a/libavcodec/mediacodec_wrapper.h b/libavcodec/mediacodec_wrapper.h
index 18d07964..97b744a7 100644
--- a/libavcodec/mediacodec_wrapper.h
+++ b/libavcodec/mediacodec_wrapper.h
@@ -235,6 +235,7 @@ struct FFAMediaCodec {

     // For encoder with FFANativeWindow as input.
     int (*signalEndOfInputStream)(FFAMediaCodec *);
+    int (*setDynamicBitrate)(FFAMediaCodec *codec, int bitrate);

     // Introduced in Android API 28
     int (*setAsyncNotifyCallback)(FFAMediaCodec *codec,
@@ -364,6 +365,11 @@ static inline int ff_AMediaCodec_signalEndOfInputStream(FFAMediaCodec *codec)
     return codec->signalEndOfInputStream(codec);
 }

+static inline int ff_AMediaCodec_setDynamicBitrate(FFAMediaCodec *codec, int bitrate)
+{
+    return codec->setDynamicBitrate(codec, bitrate);
+}
+
 static inline int ff_AMediaCodec_setAsyncNotifyCallback(FFAMediaCodec *codec,
         const FFAMediaCodecOnAsyncNotifyCallback *callback,
         void *userdata)
diff --git a/libavcodec/mediacodecenc.c b/libavcodec/mediacodecenc.c
index 507c48df..f96dc1ca 100644
--- a/libavcodec/mediacodecenc.c
+++ b/libavcodec/mediacodecenc.c
@@ -86,6 +86,7 @@ typedef struct MediaCodecEncContext {
     // Ref. MediaFormat KEY_OPERATING_RATE
     int operating_rate;
     int async_mode;
+    int64_t last_bit_rate;

     AVMutex input_mutex;
     AVCond input_cond;
@@ -399,6 +400,8 @@ static av_cold int mediacodec_init(AVCodecContext *avctx)
     int ret;
     int gop;

+    s->last_bit_rate = avctx->bit_rate;
+
     // Init async state first, so we can do cleanup safely on error path.
     ret = mediacodec_init_async_state(avctx);
     if (ret < 0)
@@ -841,6 +844,15 @@ static int mediacodec_encode(AVCodecContext *avctx, AVPacket *pkt)
     MediaCodecEncContext *s = avctx->priv_data;
     int ret;

+    if (avctx->bit_rate != s->last_bit_rate) {
+        s->last_bit_rate = avctx->bit_rate;
+        ret = ff_AMediaCodec_setDynamicBitrate(s->codec, avctx->bit_rate);
+        if (ret)
+            av_log(avctx, AV_LOG_ERROR, "Failed to set bitrate to %" PRId64 "\n", avctx->bit_rate);
+        else
+            av_log(avctx, AV_LOG_INFO, "Set bitrate to %" PRId64 "\n", avctx->bit_rate);
+    }
+
     // Return on three case:
     // 1. Serious error
     // 2. Got a packet success

