diff --git a/configure b/configure
index 5f4a5eb3..1dc5b8e4 100755
--- a/configure
+++ b/configure
@@ -7234,12 +7234,6 @@ enabled ohcodec           && { check_lib ohcodec "multimedia/player_framework/na
                                     "OH_VideoDecoder_CreateByName OH_VideoEncoder_CreateByName" \
                                     -lnative_media_vdec -lnative_media_venc -lnative_media_codecbase -lnative_media_core -lnative_window ||
                                die "ERROR: missing native_media libs"; }
-enabled openal            && { check_pkg_config openal "openal >= 1.1" "AL/al.h" alGetError ||
-                               { for al_extralibs in "${OPENAL_LIBS}" "-lopenal" "-lOpenAL32"; do
-                               check_lib openal 'AL/al.h' alGetError "${al_extralibs}" && break; done } ||
-                               die "ERROR: openal not found"; } &&
-                             { test_cpp_condition "AL/al.h" "defined(AL_VERSION_1_1)" ||
-                               die "ERROR: openal must be installed and version must be 1.1 or compatible"; }
 enabled opencl            && { check_pkg_config opencl OpenCL CL/cl.h clEnqueueNDRangeKernel ||
                                check_lib opencl OpenCL/cl.h clEnqueueNDRangeKernel "-framework OpenCL" ||
                                check_lib opencl CL/cl.h clEnqueueNDRangeKernel -lOpenCL ||

